tests/test_bookmarks.py

```diff
@@ -0,0 +1,211 @@
+
+        # Run a sync job using orchestrator
+        second_sync_record_count = self.run_and_verify_sync(conn_id)
+        second_sync_records = runner.get_records_from_target_output()
+        second_sync_bookmarks = menagerie.get_state(conn_id)
+
+        for stream in expected_streams:
+
+            with self.subTest(stream=stream):
+                # collect information for assertions from syncs 1 & 2 base on expected values
+                first_sync_count = first_sync_record_count.get(stream, 0)
+                second_sync_count = second_sync_record_count.get(stream, 0)
+
+                # collect information for assertions from syncs 1 & 2 base on expected values
+                first_sync_messages = [record.get('data') for record in first_sync_records.get(stream).get('messages')
+                                       if record.get('action') == 'upsert']
+                second_sync_messages = [record.get('data') for record in second_sync_records.get(stream).get('messages')
+                                        if record.get('action') == 'upsert']
+
+                first_bookmark_key_value = first_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+                second_bookmark_key_value = second_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+
+                if self.is_incremental(stream):
+
+                    self.assertEqual(first_bookmark_key_value, second_bookmark_key_value)
+
+                    # if the stream is child stream, then collect parent stream records for collecting ids
+                    if self.parent_child_mapping.get(stream):
+                        parent_records = [record.get('data') for record in first_sync_records.get(self.parent_child_mapping.get(stream) + 's').get('messages')
+                                        if record.get('action') == 'upsert']
+
+                    # NOTE: Sample bookmark:
+                    #   "ad_accounts": {
+                    #       "updated_at(parent_organization_id:organization-id-123)": "2020-01-01T00:00:00Z"
+                    #   }
+                    for key, value in first_bookmark_key_value.items():
+                        # partial split
+                        partial_splitted = key.split('parent_')
+
+                        # if the length of split is greater than 1, the bookmark is child stream
+                        if len(partial_splitted) > 1:
+                            splitted = partial_splitted[1].split(':')
+                            # parent value
+                            parent_value = splitted[0]
+                            # get parent record and remove last char ')' as we have splitted
+                            parent_record = splitted[1][:-1]
+
+                            # verify the parent in the bookmark is as expected
+                            self.assertEqual(parent_value, self.parent_child_mapping.get(stream) + '_id')
```

Comment:
```suggestion
                        # verify the parent in the bookmark is as expected
                        parent_stream = self.parent_child_mapping.get(stream)
                        if parent_stream:
                            self.assertEqual(f"parent_{parent_stream}_id", key)
```

Reply:
This change simplifies the logic and changes the conditional to act on the parent child map (expectation) rather than the first sync bookmark (actual).

Reply:
It will not work, as by this code we will get only some part of the bookmark and `key`. We will get assertion error:
```
AssertionError: 'parent_organization_id' != 'updated_at(parent_organization_id:7f29aa78-a928-469e-b1a8-5cfe10e1a2e9)'
- parent_organization_id
+ updated_at(parent_organization_id:7f29aa78-a928-469e-b1a8-5cfe10e1a2e9)```

Additionally, we need the parent id from the bookmark key for the assertion below. So, splitting helps us there also.